src/Tmds.DBus/CodeGen/TypeDescription.cs

```diff
@@ -150,6 +150,11 @@ private static void AddInterfaceDescription(Type type, DBusInterfaceAttribute in
 
             foreach (var member in type.GetMethods())
             {
+                // Objects that have properties do not have to have "Async" in their name and is skipped.
+                if (member.IsSpecialName)
+                {
+                    continue;
+                }
```

Comment:
I'm not sure what this is for?

Reply:
Oops, this is my first pull request (ever!) and noob mistake. I thought the pull request would only include the first commit.

But while I've got your attention, here is the problem in a small repo:
```csharp
using System;
using System.Threading.Tasks;
using Tmds.DBus;

namespace publishproperty
{
    class Program
    {
        static void Main(string[] args)
        {
            Task.Run(async () =>
            {
                using (var connection = new Connection(Address.System))
                {
                    await connection.ConnectAsync();
                    var exposed = new ExposeMe();
                    await connection.RegisterObjectAsync(exposed);

                    await Task.Delay(Int32.MaxValue);
                }
            }).Wait();
        }
    }

    class ExposeMe : IExposeMe
    {
        public string UUID {get; private set;}

        public ObjectPath ObjectPath => new ObjectPath("/com/example/object1");
    }

    [DBusInterface("com.example")]
    interface IExposeMe : IDBusObject
    {
        string UUID {get;}
    }
}
```

Am I doing it wrong? It seems the compiler is generating property accessor methods behind the scences and this is why the TypeDescription method picked it up.

Reply:
For the sake of a clean pull request, I have removed this code too. Would like your thoughts on this if you get a moment though.  Should Auto properties be avoided?